src/main/java/com/firstspringmvcapp/services/PeopleService.java

```diff
@@ -115,30 +119,24 @@ public void removeFriendById(int personId, int friendId) {
     }
 
     @Transactional
-    public void uploadProfilePicture(HttpSession session, CommonsMultipartFile file, int personId) {
-        ServletContext servletContext = session.getServletContext();
-        String path = servletContext.getRealPath("/images/pfp");
-        path = path.replace("FirstSpringMVCApp\\target\\FirstSpringMVCApp\\",
-                "FirstSpringMVCApp\\src\\main\\resources\\");
-        String filename = file.getOriginalFilename();
-
-        System.out.println(path + " " + filename);
-        System.out.println(path + "\\" + personId + ".png");
-        if (file.isEmpty() || !(filename != null && (filename.endsWith(".png") || filename.endsWith(".jpg")))) {
-            System.out.println("Your file should be valid .png image!");
-            return;
-        }
-
-        try (BufferedOutputStream bout = new BufferedOutputStream(
-                new FileOutputStream(path + "\\" + personId + ".png"))) {
-            bout.write(file.getBytes());
-            bout.flush();
-        } catch (FileNotFoundException e) {
-            e.printStackTrace();
+    public void uploadProfilePicture(MultipartFile file, int personId) {
+        try {
+            ByteArrayInputStream bais = new ByteArrayInputStream(file.getBytes());
+            MinioFileHandler.upload("picture-bucket", personId + ".png", bais);
         } catch (IOException e) {
             e.printStackTrace();
         }
 
-        peopleRepository.getOne(personId).setProfilePicturePath(personId + ".png");
+        Person updatedPerson = peopleRepository.getOne(personId);
+        if (!updatedPerson.getProfilePicturePath().equals(personId + ".png")) {
```

Comment:
the check should always starts with non null constant in similar cases to prevent NPE
if(!"personId.png".equals(updatedPerson.getProfilePicturePath())) {
}

add pictureName=personId.concat(".png") as variable and don't repeat it